File: frontend/src/types/policy.ts
export interface Policy {
  id: string;
  organizationId: string;
  name: string;
  description: string | null;
  allowedCommandPatterns: string[];
  deniedCommandPatterns: string[];
  requireApproval: boolean;
  maxConcurrentCommands: number;
  isEnabled: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePolicyRequest {
  name: string;
  description?: string;
  allowedCommandPatterns: string[];
  deniedCommandPatterns: string[];
  requireApproval: boolean;
  maxConcurrentCommands: number;
}

export interface UpdatePolicyRequest extends Partial<CreatePolicyRequest> {
  isEnabled?: boolean;
}

export interface PolicyTestResult {
  isAllowed: boolean;
  denialReason: string | null;
  requiresApproval: boolean;
  matchedPattern: string | null;
}

export interface PolicyAssignment {
  id: string;
  userId: string;
  hostId: string | null;
  hostName: string | null;
  createdAt: string;
}

export interface PolicyPreset {
  name: string;
  description: string;
  allowedCommandPatterns: string[];
  deniedCommandPatterns: string[];
  requireApproval: boolean;
  maxConcurrentCommands: number;
}

File: src/InfraLLM.Core/Models/Policy.cs
namespace InfraLLM.Core.Models;

public class Policy
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public List<string> AllowedCommandPatterns { get; set; } = [];
    public List<string> DeniedCommandPatterns { get; set; } = [];
    public int MaxConcurrentCommands { get; set; } = 5;
    public bool IsEnabled { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Organization Organization { get; set; } = null!;
    public List<UserPolicy> UserPolicies { get; set; } = [];
}
